phpBB/phpbb/captcha/plugins/recaptcha_v3.php

```diff
@@ -0,0 +1,350 @@
+			$config->set('recaptcha_v3_method', $request->variable('recaptcha_v3_method', '', true));
+
+			foreach (self::$actions as $action)
+			{
+				$config->set("recaptcha_v3_threshold_{$action}", $request->variable("recaptcha_v3_threshold_{$action}", 0.50));
+			}
+
+			$phpbb_log->add('admin', $user->data['user_id'], $user->ip, 'LOG_CONFIG_VISUAL');
+
+			trigger_error($language->lang('CONFIG_UPDATED') . adm_back_link($module->u_action));
+		}
+
+		foreach (self::$actions as $action)
+		{
+			$template->assign_block_vars('thresholds', [
+				'key'	=> "recaptcha_v3_threshold_{$action}",
+				'value'	=> $config["recaptcha_v3_threshold_{$action}"] ?? 0.5,
+			]);
+		}
+
+		$template->assign_vars([
+			'CAPTCHA_NAME'				=> $this->get_service_name(),
+			'CAPTCHA_PREVIEW'			=> $this->get_demo_template($id),
+
+			'RECAPTCHA_V3_KEY'			=> $config['recaptcha_v3_key'] ?? '',
+			'RECAPTCHA_V3_SECRET'		=> $config['recaptcha_v3_secret'] ?? '',
+
+			'RECAPTCHA_V3_DOMAIN'		=> $config['recaptcha_v3_domain'] ?? self::GOOGLE,
+			'RECAPTCHA_V3_DOMAINS'		=> [self::GOOGLE, self::RECAPTCHA],
+
+			'RECAPTCHA_V3_METHOD'		=> $config['recaptcha_v3_method'] ?? self::POST,
+			'RECAPTCHA_V3_METHODS'		=> [
+				self::POST		=> ini_get('allow_url_fopen') && function_exists('file_get_contents'),
+				self::CURL		=> extension_loaded('curl') && function_exists('curl_init'),
+				self::SOCKET	=> function_exists('fsockopen'),
+			],
+
+			'U_ACTION'					=> $module->u_action,
+		]);
+	}
+
+	/**
+	 * Create the ACP page for previewing this CAPTCHA plugin.
+	 *
+	 * @param string	$id		The module identifier
+	 * @return bool|string
+	 */
+	public function get_demo_template($id)
+	{
+		return $this->get_template();
+	}
+
+	/**
+	 * Get the template for this CAPTCHA plugin.
+	 *
+	 * @return bool|string		False if CAPTCHA is already solved, template file name otherwise
+	 */
+	public function get_template()
+	{
+		/**
+		 * @var \phpbb\config\config		$config				Config object
+		 * @var \phpbb\language\language	$language			Language object
+		 * @var \phpbb\template\template	$template			Template object
+		 * @var string						$phpbb_root_path	phpBB root path
+		 * @var string						$phpEx				php File extensions
+		 */
+		global $config, $language, $template, $phpbb_root_path, $phpEx;
+
+		if ($this->is_solved())
+		{
+			return false;
+		}
+
+		$contact = phpbb_get_board_contact_link($config, $phpbb_root_path, $phpEx);
+		$explain = $this->type !== CONFIRM_POST ? 'CONFIRM_EXPLAIN' : 'POST_CONFIRM_EXPLAIN';
+
+		$domain = $config['recaptcha_v3_domain'] ?? self::GOOGLE;
+		$render = $config['recaptcha_v3_key'] ?? '';
+
+		$template->assign_vars([
+			'CONFIRM_EXPLAIN'		=> $language->lang($explain, '<a href="' . $contact . '">', '</a>'),
+
+			'RECAPTCHA_ACTION'		=> self::$actions[$this->type] ?? reset(self::$actions),
+			'RECAPTCHA_KEY'			=> $config['recaptcha_v3_key'] ?? '',
+			'U_RECAPTCHA_SCRIPT'	=> sprintf('//%s/recaptcha/api.js?render=%s', $domain, $render),
```

Comment:
Please use the same method as in language files, e.g.
```suggestion
			'U_RECAPTCHA_SCRIPT'	=> sprintf('//%1$s/recaptcha/api.js?render=%2$s', $domain, $render),
```